src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -178,6 +186,151 @@ protected boolean doCreateTableIfNotExists(
         return tableState == TableResult.State.ACTIVE;
     }
 
+    protected void doCheckExistingTable(NosqlEntityInformation<?, ?> entityInformation) {
+        final String columnsField = "fields";
+        final String columnNameField = "name";
+        final String columnTypeField = "type";
+        final String shardField = "shardKey";
+        final String primaryField = "primaryKey";
+        final String ttlField = "ttl";
+        final String identityField = "identity";
+
+        TableResult tableResult = doGetTable(entityInformation);
+        /*If table already exist in the database compare and throw error if
+          mismatch*/
+        if (tableResult != null) {
+            MapValue jsonSchema = JsonUtils.createValueFromJson(
+                            tableResult.getSchema(),
+                            new JsonOptions().setMaintainInsertionOrder(true)).
+                    asMap();
+
+            Map<String, FieldValue.Type> shardKeys = entityInformation.
+                    getShardKeys();
+            Map<String, FieldValue.Type> nonShardKeys = entityInformation.
+                    getNonShardKeys();
+
+            //lower case maps
+            Map<String, FieldValue.Type> caseShardKey = new LinkedHashMap<>();
+            shardKeys.forEach((k, v) -> caseShardKey.put(k.toLowerCase(), v));
+
+            Map<String, FieldValue.Type> caseNonShardKey =
+                    new LinkedHashMap<>();
+            nonShardKeys.forEach((k, v) -> caseNonShardKey.put(k.toLowerCase(), v));
+
+
+            ArrayValue columns = jsonSchema.get(columnsField).asArray();
+            if (columns != null) {
+                //check number of columns are same
+                if (columns.size() != shardKeys.size() + nonShardKeys.size() + 1) {
+                    throw new IllegalArgumentException(
```

Comment:
I think it's important to accumulate all the mismatches in a list or string and when the error is thrown all the mismatches should be printed. This way the user can fix the problem in one single iteration. Something like:
IllegalArgumentException: The following mismatches have been found between the entity definition Customer class and the existing store table CustomeTable:
  - key field type mismatch: entity int, table string.
  - order of composite primary key: entity shows "city, country" table show "country, city"
  - etc
To fix this error, Customer class has to match the table Customer or use NosqlTable.name annotation to use a different table.

Reply:
I'll try to add as much information possible. May not be exactly same as an example provided by you.

Reply:
Updated. Now error look like below.
```
java.lang.IllegalArgumentException: The following mismatch have been found between the entity class com.oracle.nosql.spring.data.test.composite.TestTableCreation$MismatchEntity definition and the existing table MismatchEntity in the database:
Shard primary keys mismatch: table={id1 string}, entity={id1 integer,id3 integer}.
Non-shard primary keys mismatch: table={id2 string}, entity={id4 integer}.
Non-primary key columns mismatch:table={kv_json_ long,age integer}, entity={kv_json_ json}.
To fix this errors, either make the entity class to match the table definition or use NosqlTable.name annotation to use a different table.

```